fix(core): Fix MySQL tests timeout issues

despairblue opened the pull request:
## Summary

This fixes the remaining MySQL test failing.

Additionally I identified some tests that would not shut down properly, mainly because they don't close the connection to the db, while others depend on the WaitTracker which does is not terminated implicitly and so will keep the event loop busy.

For the test db it would be nice to have this work implicitly like the test server does it:
https://github.com/n8n-io/n8n/blob/57c95dcf2d0d2af208fe5f59ba14cd09678466cf/packages/cli/test/integration/shared/utils/testServer.ts#L70-L285

But that would be a bigger refactor which I don't want to include into feature/rbac, nor want to do it on master now for the fear or conflicts with feature/rbac.

## Related tickets and issues

https://linear.app/n8n/issue/PAY-1569/mysql-tests-failing

## Review / Merge checklist

- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))

despairblue committed:
fix flaky test

despairblue committed:
fix flaky test

despairblue committed:
fix flaky test

despairblue committed:
fix flaky test

despairblue committed:
fix flaky test

despairblue committed:
fix flaky test

despairblue committed:
fix flaky test

despairblue commented on .github/workflows/ci-postgres-mysql.yml:
```diff
@@ -5,8 +5,6 @@ on:
     - cron: '0 0 * * *'
   workflow_dispatch:
   pull_request:
-    paths:
-      - packages/cli/src/databases/**
```
Just temporary here to test the workflow.

netroy replied:
instead of removing this line, I think we should add `.github/workflows/ci-postgres-mysql.yml` to the list

netroy commented on .github/workflows/ci-postgres-mysql.yml:
```diff
@@ -92,7 +90,7 @@ jobs:
       - name: Test MySQL
         working-directory: packages/cli
-        run: pnpm test:mysql
+        run: pnpm test:mysql --testTimeout 20000
```
I've seen this timeout on `master` at least twice last week. Can you make this PR against `master` instead please 🙏🏽

despairblue replied:
Here you go: https://github.com/n8n-io/n8n/pull/9332

despairblue committed:
fix test failing only on mysql

If we don't stringify `CredentialEntity.data` MySQL throws this error:

    QueryFailedError: Unknown column 'accessToken' in 'field list'

Encrypting the data achieves this.
